maven-model-transform/src/main/java/org/apache/maven/model/transform/pull/NodeBufferingParser.java

```diff
@@ -38,44 +37,54 @@
 
     private final List<Event> buffer = new ArrayList<>();
 
-    private final String nodeName;
+    private final ArrayList<String> stack = new ArrayList<>();
```

Comment:
Shouldn't this variable be of type `List<String>`? "Program against interfaces, not implementations."

Reply:
"... on API edges", yes, but this is totally and fully internal matter/state of implementation, where that mantra above makes sense only in increasing import lines count, nothing more. I don't see this internal `stack` member exposed anywhere, so this is IMHO much better than saying `SuperType stack = new Type` that again, results in +1 import line only, nothing more.

Reply:
@cstamas I don't think this argument really holds in this case. The `NodeBufferingParser` also defines another abstract method which is defined by `void process( List<Event> buffer )`, so using a `List` instead of an `ArrayList` actually removes one import in all implementing classes.